### ISSUE TITLE
Migrate to a New Model for Storing Vaccination Details of a Patient

### ISSUE THREAD

**Issue by gigincg** [coronasafe/care_fe#7876]
Currently CARE only supports storage of Vaccination details related to COVID. We need to generalise this and create a new model to store all vaccination details of a patient. Thereby removing these fields from the PatientRegistration Table. The Vaccination model should store comprehensive vaccination information to track a patient’s vaccination history effectively.

- [ ] Backend Changes to create the new model PatientVaccination & related viewsets and serializers with Suggestive Fields:
	•	patient (ForeignKey to PatientRegistration)
	•	vaccine_name (choices from vaccineEnum)
	•	date_of_vaccination (DateField)
	•	dose_number (PositiveIntegerField, default 1)
	•	vaccination_center (CharField, max_length=255)
	•	batch_number (CharField, max_length=100)
	•	created_by (ForeignKey to User)
	•	last_edited_by (ForeignKey to User)
- [ ] Create Mockups/Suggestion for handling the same in the UI.
- [ ] Frontend Changes in PatientRegistration Form, Patient Profile, Link in Patient Dashboard for Vaccination Details

**Comment by hrit2773** [comment]
@gigincg @rithviknishad  can you assign this to me

**Comment by gigincg** [comment]
Hey @hrit2773 Can you outline your plan for solving this issue?

**Comment by Atharva-Kanherkar** [comment]
@gigincg I would love to work on the issue too. I have worked with creating models for databases, mainly in typescript and graphQL. I would love to contribute.

**Comment by gigincg** [comment]
@Atharva-Kanherkar The Care backend is in Django/Python. If you wanna take this up, please do explain your approach. 

Since this is a core issue, we can only assign this issue post design discussions. If you would prefer to discuss over our community calls, you can find the invites in our slack

**Comment by hrit2773** [comment]
![Screenshot 2024-05-28 163450](https://github.com/coronasafe/care_fe/assets/128292557/b5a5eb94-5199-4559-9641-adccb4af5bf0)
@gigincg the form may look like this. Once the user enters "Yes" for the question "Is patient Vaccinated?" this will be visible and vaccine name will be a search and select form field

**Comment by hrit2773** [comment]
![Screenshot 2024-05-28 170621](https://github.com/coronasafe/care_fe/assets/128292557/688e2c5e-90a7-451b-919b-a779222b01ca)
@gigincg we can create a table kind of layout instead of just vaccine name, last vaccinated on, number of doses. This would show all the vaccination details of the patient for multiple vaccines he/she has taken.

**Comment by hrit2773** [comment]
@gigincg and for backend new model Patient Vaccination can be created along with its serializers and viewsets. So backend can  be done as specified along with proper validations. I can start working on both frontend and backend once this is approved

**Comment by hrit2773** [comment]
@gigincg and for vaccine name choices we can add new vaccine names. Will find the names from web

**Comment by hrit2773** [comment]
@gigincg any other changes from your side? Plz let me know

**Comment by gigincg** [comment]
@hrit2773 Assigning you the issue. 

Your appriach sounds good. 

Given the backend changes and migrations required, file a PR to the github.com/coronasafe/docs repo explaining the Model Changes and Migrations you will be doing before working on dev. Also please confirm if you will be making both the FE and BE PRs required.

**Comment by hrit2773** [comment]
@gigincg Yes I will be doing both frontend and Backend PR's and will raise a draft PR explaining the model changes

**Comment by hrit2773** [comment]
@gigincg https://github.com/coronasafe/docs/pull/27 can you just review this. I have updated both the model changes in the backend and the frontend mockups

**Comment by gigincg** [comment]
@hrit2773 Reviewed the docs, take a look at the notes and go ahead with the implementation (Backend and Frontend PRs)

**Comment by hrit2773** [comment]
@gigincg https://github.com/coronasafe/care/pull/2275

**Comment by hrit2773** [comment]
@gigincg can you plz review the frontend and backend changes. I have completed all the functionalities. Any small UI changes required? will do them immediately. cc: @rithviknishad

**Comment by gigincg** [comment]
@vigneshhari This will be covered the Health Details APIs right?

**Comment by nihal467** [comment]
not a requirement based on the current develop

**Comment by bodhish** [comment]
Closing in favour of FHIR Vac
